projects/output-and-error-handling/client/main.go

```diff
@@ -0,0 +1,73 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"strconv"
+	"time"
+)
+
+func main() {
+	// Connect to the server and getting a response
+	resp, err := http.Get("http://localhost:8080")
+	// Show error message if connection is not established
+	if err != nil {
+		handleError()
+	}
+
+	// Close the response body after been fully read
+	defer resp.Body.Close()
+
+	// Read response body and store it in a var
+	body, err := io.ReadAll(resp.Body)
+
+	// Show error message if we cannot read the response body
+	if err != nil {
+		handleError()
+	}
+	// Convert response body from binary to string
+	sb := string(body)
```

Comment:
This logic is fine, but it only gets used if the status is 200. So in every other scenario (500, 429) it's an unnecessary operation.